src/_P098_CacheReader.ino

```diff
@@ -0,0 +1,478 @@
+    (*this->fileNr) = 0;
+    (*this->sampleIndex) = 0;
+    }
+    //fnr++;
+    
+    delete[] fname;
+  }
+};
+
+struct P098_data_struct : public PluginTaskData_base {
+  Cache_t *cache = nullptr;
+  P098_data_struct() {}
+  ~P098_data_struct(){
+    reset();
+  }
+  void reset(){
+    if (cache != nullptr){
+      delete cache;
+      cache = nullptr;
+    }
+  }
+  bool init(){
+    reset();
+    cache = new Cache_t();
+    return isInitialized();
+  }
+  bool isInitialized() const {
+    return cache != nullptr;
+  }
+};
+
+boolean Plugin_098(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+      {
+        Device[++deviceCount].Number = PLUGIN_ID_098;
+        Device[deviceCount].Type = DEVICE_TYPE_DUMMY;
+        Device[deviceCount].VType = SENSOR_TYPE_SINGLE;
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = false;
+        Device[deviceCount].DecimalsOnly = true;
+        Device[deviceCount].ValueCount = 4;
+        Device[deviceCount].SendDataOption = true;
+        Device[deviceCount].TimerOption = true;
+        Device[deviceCount].GlobalSyncOption = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICENAME:
+      {
+        string = F(PLUGIN_NAME_098);
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+      {
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_098));
+        break;
+      }
+
+    case PLUGIN_WEBFORM_LOAD:
+      {
+        sensorTypeHelper_webformLoad_allTypes(event, 0);
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+        sensorTypeHelper_saveSensorType(event, 0);
+        success = true;
+        break;
+      }
+
+    case PLUGIN_INIT:
+      {
+        initPluginTaskData(event->TaskIndex, new P098_data_struct());
+        P098_data_struct *P098_data =
+          static_cast<P098_data_struct *>(getPluginTaskData(event->TaskIndex));
+        P098_data->init();
+        P098_data->cache->initialize(event);
+        success = true;
+        break;
+      }
+    case PLUGIN_EXIT: {
+      clearPluginTaskData(event->TaskIndex);
+      success = true;
+      break;
+    }
+    case PLUGIN_READ:
+      {
+        success = true;
+        break;
+      }
+    case PLUGIN_ONCE_A_SECOND:
+    {
+      P098_data_struct *P098_data =
+        static_cast<P098_data_struct *>(getPluginTaskData(event->TaskIndex));
+      bool initialized = P098_data->isInitialized();
+
+      float killswitch = (*P098_data->cache->sampleCount);
+
+      // Keep during alpha to avoid reflashing with usb
+      if (killswitch > 0){
+        //(*P098_data->cache->sampleCount)--;
+
+        if (initialized){
+          for (int i = 0 ; i < 2 ; i++){
+            P098_data->cache->readSample(event);
```

Comment:
Why read 2 samples per run?
What makes "2" the best choice here?  (Maybe it should be a setting of the plugin)